File: IT/Programmazione funzionale/Esercizi/Python/esercizio21.py
# file esercizio21.py

from __future__ import print_function
import math

def compose(g, f):
	return lambda x : g(f(x))

f = lambda x : math.e**x
g = lambda y : y + 1

# funzione composta (g o f)(x) = e^x + 1

gof = compose(g, f)

print(gof)
print("e^3 + 1 = {:.2f}".format(gof(3)))

File: IT/Programmazione funzionale/Esercizi/Java/Esercizio21.java
/* file Esercizio21.java */

import java.util.Locale;
import java.util.function.Function;

public class Esercizio21 {
	public static void main(String [] args) {
		Function<Double, Double> f = (x) -> Math.pow(Math.E, x);
		Function<Double, Double> g = (y) -> y + 1;

		// funzione composta (g o f)(x) = e^x + 1
		Function<Double, Double> gof = g.compose(f);

		System.out.println("e^3 + 1 = " + String.format(Locale.ROOT, "%.4f", gof.apply(3.0)));
	}
}

